### PR TITLE
Basic test for remote method

### PR DESCRIPTION
**Pull request by listepo**
@digitalsadhu this PR to repair remote methods. This is a basic test. You can fix if you see where I'm wrong. I'm not very well studied this framework

### CODE REVIEW

File: test/remoteMethods.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    );
+    app.model(Post);
+    app.use(loopback.rest());
+    JSONAPIComponent(app);
+  });
+
+  describe('status codes', function () {
+    it('POST /models should return a 201 CREATED status code', function (done) {
+      request(app).post('/posts')
```
Comment (digitalsadhu) [inline test/remoteMethods.test.js]: This should be to /posts/greet right?

File: test/remoteMethods.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      }
+    );
+    app.model(Post);
+    app.use(loopback.rest());
+    JSONAPIComponent(app);
+  });
+
+  describe('status codes', function () {
+    it('POST /models should return remote method message', function (done) {
+      request(app).post('/posts')
```
Comment (digitalsadhu) [inline test/remoteMethods.test.js]: This should be a GET to /posts/greet?

Comment (digitalsadhu) [inline test/remoteMethods.test.js]: To /posts/greet/John I think.

File: test/remoteMethods.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        .send({'msg': 'John'})
+        .set('Content-Type', 'application/json')
+        .expect(200)
+        .end(function (err, res) {
+          expect(err).to.equal(null);
+          expect(res.body).to.equal('Greetings... John!');
```
Comment (digitalsadhu) [inline test/remoteMethods.test.js]: res.body.greeting I think.

File: test/remoteMethods.test.js (hunk 2 above)
Comment (listepo) [inline test/remoteMethods.test.js]: @digitalsadhu 
POST to /posts/greet
see https://docs.strongloop.com/display/public/LB/Remote+methods

Comment (listepo) [inline test/remoteMethods.test.js]: but I think it works the same for GET

**Comment by digitalsadhu** [comment]
Nice one @listepo ! I had to fix both serialize and deserialize to get that test to pass. It may have fixed the user model issues as well. Can you check?

**Comment by listepo** [comment]
@digitalsadhu, good work. But I think need include support JSONAPI if that indicated in the header. What do you think about this? I tried to express the idea in the code, but has not got to do to make things work.

**Comment by listepo** [comment]
Ember Data 1.13.x default sends only ```Accept: application/vnd.api+json``` without ```Content-Type application/vnd.api+json```  I think it is necessary to take this into account

**Comment by digitalsadhu** [comment]
I agree however for your tests see this issue i created for supertest/superagent:
https://github.com/visionmedia/superagent/issues/753
Thats why I've been using application/json in the tests.

Also, I don't think it will be possible to make this work properly due to the overwriting of relationship methods here: 
https://github.com/digitalsadhu/loopback-component-jsonapi/blob/master/lib/patch.js#L39-L190

I think for now we have to make all of the core methods always send out the `application/vnd.api+json` Content-Type header if they user has the loopback-component-jsonapi installed.
I plan on trying to find out from the strongloop guys if there might be another way to patch the relationship methods so that we can make the changes on a per request basis and switch on and off based on the Accept header.

Does that sound ok?

**Comment by digitalsadhu** [comment]
@listepo ^

**Comment by listepo** [comment]
Yes thank you:)

**Pull request merged**